deeppavlov/models/go_bot/trippy_preprocessing.py

```diff
@@ -0,0 +1,967 @@
+# Copyright 2021 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+
+### Implements Preprocessing functions to turn DSTC2 into TripPy readable format ###
+
+### Funcs ###
+
+import string
+import re
+from difflib import ndiff
+from logging import getLogger
+from typing import Dict, Any, List, Optional, Union, Tuple
+
+import six
+import numpy as np
+import torch
+
+logger = getLogger(__name__)
+
+
+class DSTExample(object):
```

Comment:
this should be implemented as dataset_reader or dataset_iterator

Reply:
The whole preprocessing file?

Reply:
The problem is though that we also feed it through the trippy preprocessing when you just call the go bot with e.g. bot(["Hi")
This data does not pass by the dataset_reader or dataset_iterator so we couldn't preprocess it.

We could possibly implement as its own chainer element before the trippy model, but then I'm
a) Not sure if they get called when we call the model?
b) There would be no real benefit to the decoupling in the chain, as you would never use the trippy preprocessing without also using the trippy